src/meta/api/src/schema_api_impl.rs

```diff
@@ -1043,41 +1019,48 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
     }
 
     #[tracing::instrument(level = "debug", ret, err, skip_all)]
-    async fn get_indexes_by_table_id(
+    async fn list_indexes(
         &self,
-        req: ListIndexByTableIdReq,
-    ) -> Result<Option<Vec<(IndexId, IndexMeta)>>, KVAppError> {
+        req: ListIndexesReq,
+    ) -> Result<Vec<(u64, String, IndexMeta)>, KVAppError> {
         debug!(req = debug(&req), "SchemaApi: {}", func_name!());
 
-        // get index id list from _fd_index_id_list/<tenant>/<table_id>
-        let index_id_list_key = IndexIdListKey {
-            tenant: req.tenant.clone(),
-            table_id: req.table_id,
-        };
-
-        let (_, index_id_list): (_, Option<IndexIdList>) =
-            get_pb_value(self, &index_id_list_key).await?;
+        // Get index id list by `prefix_list` "<prefix>/<tenant>"
+        let prefix_key = kvapi::KeyBuilder::new_prefixed(IndexNameIdent::PREFIX)
+            .push_str(&req.tenant)
+            .done();
+
+        let id_list = self.prefix_list_kv(&prefix_key).await?;
+        let mut id_name_list = Vec::with_capacity(id_list.len());
+        for (key, seq) in id_list.iter() {
+            // Safety: the key is derived from the `prefix_key` above.
+            let name_ident = IndexNameIdent::from_str_key(key).unwrap();
```

Comment:
There may still be incorrect data responded. It would be better to return an error, such as `MetaNetworkError::InvalidReply`